### PR TITLE
DAML/JavaScript: Make naming in docs consistent

### PR DESCRIPTION
**Pull request by hurryabit**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/5475)

### CODE REVIEW

File: docs/source/app-dev/bindings-ts/daml2js.rst [hunk 1]
```diff
@@ -1,18 +1,12 @@
 .. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-DAML to JavaScript code generation
-##################################
+DAML/JavaScript Code Generator
```
Comment (cocreature) [inline docs/source/app-dev/bindings-ts/daml2js.rst:4]: This sounds like you can generate either DAML or JavaScript. I find `to` to be easier to understand. Or just `JavaScript` codegentor.

File: docs/source/app-dev/bindings-ts/index.rst [hunk 2]
```diff
@@ -1,13 +1,14 @@
 .. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-TypeScript bindings
-###################
+DAML/JavaScript Client Libraries
```
Comment (cocreature) [inline docs/source/app-dev/bindings-ts/index.rst:4]: I’m not sure adding `DAML` in the name really helps. This is hosted on `docs.daml.com`. Anything that isn’t about DAML wouldn’t make sense here anyway.

File: docs/source/index.rst [hunk 3]
```diff
@@ -42,7 +42,6 @@ DAML SDK documentation
    :caption: Building applications
 
    app-dev/app-arch
-   daml2js/index
```
Comment (cocreature) [inline docs/source/index.rst:45]: I guess this is intentional to only have a single entry point? Sounds reasonable to me.

Comment (hurryabit) [reply docs/source/index.rst:45]: Yes, that's intended for exactly that reason. 😃

**Pull request merged**